### PR TITLE
Add a resource: Discovery Engine Data Store

### PR DESCRIPTION
**Pull request by jialei-chen**
```release-note:new-resource
`google_discovery_engine_data_store`
```

### CODE REVIEW

**Comment by jialei-chen** [comment]
@roaks3 Hi Ryan, it seems I cannot directly modify the reviewer list to add you there. So here is a friendly reminder.

Given [the context](https://github.com/GoogleCloudPlatform/magic-modules/pull/9682#issuecomment-1899331791), I finished as our discussion. Here is a brief summary: all codes remain the same as that in https://github.com/GoogleCloudPlatform/magic-modules/pull/9682 except for the following two changes:
1. Updated API version from `v1alpha` to `v1`.
2. Supported regional APIs: from `discoveryengine.googleapis.com` to `{{location}}-discoveryengine.googleapis.com`.

Thanks!

File: mmv1/products/discoveryengine/DataStore.yaml [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+--- !ruby/object:Api::Resource
+name: 'DataStore'
+description: |
+  Data store is a collection of websites and documents used to find answers for
+  end-user's questions in Discovery Engine (a.k.a. Vertex AI Search and
+  Conversation).
+references: !ruby/object:Api::Resource::ReferenceLinks
+  guides:
+    'Create a search data store': 'https://cloud.google.com/generative-ai-app-builder/docs/create-data-store-es'
+  api: 'https://cloud.google.com/generative-ai-app-builder/docs/reference/rest/v1alpha/projects.locations.collections.dataStores'
```
Comment (Pratster95) [inline mmv1/products/discoveryengine/DataStore.yaml:23]: @jialei-chen The reference doc points to alpha API

Comment (jialei-chen) [reply mmv1/products/discoveryengine/DataStore.yaml]: Thanks. Updated.

**Review by roaks3 — APPROVED** [review]
LGTM, and things match what was previously merged.

The new location-based subdomain LGTM too, but it has been a while since we've added one of these, and I'm unsure if there are other implications. Deferring to @rileykarson to make sure that piece looks alright.

**Review by rileykarson — APPROVED** [review]
Nothing seems off w/ it. We'll need to make sure `location` is always required in the product or we ensure that parent fields include it if it isn't, as the errors where it's missing are unintuitive. No impact here, where that's not a concern yet.

@roaks3 leaving merge to you as I've looked at probably 10 lines in this PR.

**Comment by roaks3** [comment]
Just did one last spot check and things looked good. Merging now.

**Pull request merged**